app/services/usps_in_person_proofing/transliterable_validator_helper.rb

```diff
@@ -0,0 +1,88 @@
+module UspsInPersonProofing
+  class TransliterableValidatorHelper
+    SUPPORTED_FIELDS = [
+      :first_name,
+      :last_name,
+      :address1,
+      :address2,
+      :city,
+    ].freeze
+
+    def initialize(transliterator = Transliterator.new)
+      @transliterator = transliterator
+    end
+
+    # Validate that transliteratable fields from the state ID form
+    # can be transliterated to contain acceptable characters.
+    #
+    # @param [Hash] fields Name/value pairs of fields to validate
+    # @option fields [String,#to_s] :first_name
+    # @option fields [String,#to_s] :last_name
+    # @option fields [String,#to_s] :address1
+    # @option fields [String,#to_s] :address2
+    # @option fields [String,#to_s] :city
+    # @return [Hash,nil] Field names mapped to error messages, or nil if there were no issues
+    def validate(fields)
+      validation_result = {}
+      fields.slice(*SUPPORTED_FIELDS).each do |key, value|
+        # Get transliterated value
+        result = @transliterator.transliterate(value)
+        transliterated = result.transliterated
+
+        # Remove question marks corresponding with unsupported chars
+        # for transliteration
+        result.unsupported_chars.each do |chr|
+          transliterated = transliterated.sub(Transliterator::REPLACEMENT, '')
+        end
+
+        # Scan for unsupported chars for the field
+        additional_chars = transliterated.scan(ADDITIONAL_UNSUPPORTED_CHARS[key])
+
+        # Create sorted list of unique unsupported characters
+        unsupported_char_list = (result.unsupported_chars + additional_chars).sort.uniq
+
+        # Use unsupported character list to generate error message
+        unless unsupported_char_list.empty?
+          validation_result[key] = send(
+            UNSUPPORTED_CHARS_TRANSLATION[key],
+            unsupported_char_list,
+          )
```

Comment:
I think this might be a good case for a `case`/`when` instead of a hash lookup, so that we don't have to resort to `send`

```suggestion
          validation_result[key] = case key
          when :first_name, :last_name
            translate_name_chars_error(unsupported_char_list)
          when :address1, :address2, :city
            translate_address_chars_error(unsupported_char_list)
          else
            raise "Unknown attribute #{key}"
```

Reply:
I'd like to see your response to [my prior comment](https://github.com/18F/identity-idp/pull/7938#discussion_r1128851186) before moving forward with this.

Reply:
The specification of fields vs messages has been updated per your refactor suggestion.